### PR TITLE
feat: argument support

### PR DESCRIPTION
**Pull request by CovertDuck**
Added basic argument support, might expand upon it later on.
See #9.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,19 +1,39 @@
...
     data: dict = read_json()
     html: str = convert_json_to_html(data)
-    write_html(html)
+    write_html(html, args.output)
 
-    print("Done!")
+    log("Done!\n")
+
+
+def log(message: str, bold: bool = False) -> None:
```
Comment (ivnvxd) [inline main.py:27]: If adding logging consider using the `logging` module instead of a custom `log()` function.

File: main.py [hunk 2]
```diff
@@ -1,19 +1,39 @@
...
 
 def main() -> None:
+    global SILENT
+    parser = argparse.ArgumentParser(description="Reads Arc Browser JSON data, converts it to HTML, and writes the output to a specified file.")
+    parser.add_argument('-s', '--silent', action='store_true', help='Silence output')
+    parser.add_argument('-o', '--output', type=Path, required=False, help='Specify the output file path')
+
+    args = parser.parse_args()
+    SILENT = args.silent
+    args.output if args.output else None
```
Comment (ivnvxd) [inline main.py:18]: Simply pass `args.output` to the function. It's already `None` by default if not provided, so this line is redundant.

**Review by ivnvxd — COMMENTED** [review]
Great PR. Thank you! Please consider making some minor changes.

File: main.py (hunk 2 above)
Comment (CovertDuck) [reply main.py:18]: Nice catch.

**Comment by CovertDuck** [comment]
Cleaned up and added the logging import. :)

**Comment by CovertDuck** [comment]
Even more goodies! :D
I removed the commented-out block, and began looking up if pinned spaces can be managed a little more cleanly...

**Comment by ivnvxd** [comment]
To be honest, I would leave the output to the console by default. It's clearer for those who are not familiar with console and just want to export bookmarks and forget about this script once and for all.

And if you add so much functionality, it would be right to add a description in README.md. For example:

Usage
Run the main.py script from the command line:

```sh
python3 main.py [options]

# or if there is an error:
python main.py [options]
```

Features
The script supports various command-line options for enhanced functionality:

-s, --silent: Silence output
-o, --output [PATH]: Specify the output file path
-v, --verbose: Increase verbosity level (e.g., -v, -vv, -vvv)
--version: Print the git version and exit

Example usage:

```sh
python3 main.py -v -o my_bookmarks.html
```

**Comment by CovertDuck** [comment]
Now printing `INFO` logs unless silenced. :)
I tweaked the README.md and listed the new arguments.
Also added a short GIF example as well, using [VHS](https://github.com/charmbracelet/vhs).

Didn't push the VHS file, no need to bloat the repository with it IMO.

```
Output example.gif

Require python3

Set Shell "bash"
Set FontSize 18
Set Width 900
Set Height 600
Set Theme "Catppuccin Macchiato"

Type "python3 main.py -v -o my_bookmarks.html"
Sleep 1500ms Enter

# Adding space between the two commands
Sleep 2000ms

Type "file my_bookmarks.html"
Sleep 500ms Enter

Sleep 5s
```

**Review by ivnvxd — APPROVED** [review]

**Pull request merged**